### ISSUE TITLE
Experimental rule 'td-has-header' does not reflect behavior

### ISSUE THREAD

**Issue by aardrian** [dequelabs/axe-core#3763]
### Product

axe Extension

### Product Version

v4.36.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
Identify when tables larger than 3 × 3 contain cells with no associated table header. Generally, flag missing `<th>`s (or `role="rowheader"` or `role="columnheader"`).

#### Actual
Tables at least up to 5 × 5 that are missing `<th>`s (or `role="rowheader"` or `role="columnheader"`) are not flagged as in violation of the rule.

In addition, tables pulled from the [Axe integration tests](https://github.com/dequelabs/axe-core/blob/develop/test/integration/rules/td-has-header/td-has-header.html) that appear to be in violation of this rule are also not flagged, specifically those coded with `id="fail1"` and `id="fail2"`.

#### How to Reproduce
1. Visit https://cdpn.io/aardrian/debug/RwyEYEP;
2. Use the Axe DevTools browser extension to scan the page;
3. Observe no errors nor warnings are raised;
4. Observe the last two tables (dotted red borders) are also not flagged.

The pen in editor view: https://codepen.io/aardrian/pen/RwyEYEP

#### Additional context
I am testing with the latest version of the Axe DevTools browser extension, which uses axe-core 4.4.2. I know this is not the latest axe-core release, but I have no method to test axe-core 4.5.

* [td-has-header](https://dequeuniversity.com/rules/axe/4.4/td-has-header) in axe-core 4.4
* [td-has-header](https://dequeuniversity.com/rules/axe/4.5/td-has-header) in axe-core 4.5

**Comment by straker** [comment]
Thanks for the issue. Experimental rules are turned off by default, so the Devtools Browser extension will not run them. In order to run experimental rules you'll need to use axe-core directly on the page and use `runOnly`

```js
const results = await axe.run({ runOnly: 'td-has-header' })
```

Doing this shows there's 12 violations on the page.

**Comment by aardrian** [comment]
Well, that clarifies a ton since I was unaware the experimental rules were not available to the extension. Thank you!

Closing this since it seems the rule runs fine but my expectation was off.

**Comment by straker** [comment]
I can see why that would be, seems our doc is wrong https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md#experimental-rules. Experimental rules were enabled by default in axe-coconut, but not in the Devtools Extension. I'll update the docs to make that more clear.

**Comment by aardrian** [comment]
I would appreciate that, since I sincerely thought I had done my legwork and just mis-read everything (which I may have done anyway).
